src/main/java/com/autotune/analyzer/workerimpl/BulkJobManager.java

```diff
@@ -216,31 +238,85 @@ public void run() {
+                                                                    || (field.getDeclaringClass() == ContainerAPIObject.class && (field.getName().equals("metrics")));
+                                                        }
+
+                                                        @Override
+                                                        public boolean shouldSkipClass(Class<?> clazz) {
+                                                            return false;
+                                                        }
+                                                    };
+                                                    Gson gsonObj = new GsonBuilder()                        //Todo move this to common code
+                                                            .disableHtmlEscaping()
+                                                            .setPrettyPrinting()
+                                                            .enableComplexMapKeySerialization()
+                                                            .registerTypeAdapter(Date.class, new GsonUTCDateAdapter())
+                                                            .registerTypeAdapter(AnalyzerConstants.RecommendationItem.class, new RecommendationItemAdapter())
+                                                            .registerTypeAdapter(DeviceDetails.class, new DeviceDetailsAdapter())
+                                                            .setExclusionStrategies(strategy)
+                                                            .create();
+                                                    experiment.getApis().getRecommendations().setResponse(gsonObj.fromJson(recommendationResponseCode.getResponseBody().toString(), List.class));
                                                     if (recommendationResponseCode.getStatusCode() == HttpURLConnection.HTTP_CREATED) {
-                                                        experiment.getRecommendations().setStatus(NotificationConstants.Status.PROCESSED);
+                                                        experiment.setStatus(NotificationConstants.Status.PROCESSED);
+                                                        // if kafka is enabled, push the recommendation response in the recommendations topic
+                                                        if(KruizeDeploymentInfo.is_kafka_enabled) {
+                                                            String processedJobDataJson = buildKafkaResponse(jobData, experiment_name, KruizeConstants.KAFKA_CONSTANTS.RECOMMENDATIONS_TOPIC, null);
+                                                            try {
+                                                                //TODO: need to add metrics response in case of kafka failure
+                                                                initiateKafkaCall(KruizeConstants.KAFKA_CONSTANTS.RECOMMENDATIONS_TOPIC, processedJobDataJson);
```

Comment:
If Kafka call is successful then set status to  PUBLISHED
experiment.setStatus(NotificationConstants.Status.PUBLISHED);

If ti failed 


experiment.setStatus(NotificationConstants.Status.PUBLISHE_FAILED);


Please catch error and pass it back and set error message.